External secrets: add readme (#3931)
kyma-project/test-infra: prow/cluster/resources/external-secrets/README.md

# External secrets

## Overview

Kubernetes secrets are synchronized with GCP Secret Manager using [Kubernetes External Secrets](https://github.com/external-secrets/kubernetes-external-secrets).

# Installation

Follow these steps to install `kubernetes-external-secrets` on an untrusted cluster in the `external-secrets` namespace:

1. Create the `external-secrets` namespace. Run `kubectl create namespace external-secrets`.
2. Add the `sa-secret-manager-untrusted` secret containing credentials for GCP service account with permission to access secrets.
3. Add `external-secrets` Helm repo. Run `helm repo add external-secrets https://external-secrets.github.io/kubernetes-external-secrets/`.

> 4. Install `external-secrets/kubernetes-external-secrets` Helm chart. Run `helm install -f prow/cluster/resources/external-secrets/values_untrusted.yaml -n external-secrets kubernetes-external-secrets external-secrets/kubernetes-external-secrets`.

Reviewer: ```suggestion
4. Install  the`external-secrets/kubernetes-external-secrets` Helm chart. Run `helm install -f prow/cluster/resources/external-secrets/values_untrusted.yaml -n external-secrets kubernetes-external-secrets external-secrets/kubernetes-external-secrets`.
```

> # Configuration

Reviewer: ```suggestion

# Configuration
```

Reviewer: Please add an empty line above the header.

> # External secrets

Reviewer: ```suggestion
# External Secrets
```

## Overview

> Kubernetes secrets are synchronized with GCP Secret Manager using [Kubernetes External Secrets](https://github.com/external-secrets/kubernetes-external-secrets).

Reviewer: ```suggestion
Kubernetes Secrets are synchronized with GCP Secret Manager using [Kubernetes External Secrets](https://github.com/external-secrets/kubernetes-external-secrets).
```

# Installation

Follow these steps to install `kubernetes-external-secrets` on an untrusted cluster in the `external-secrets` Namespace:

1. Create the `external-secrets` Namespace. Run `kubectl create namespace external-secrets`.

> 2. Add the `sa-secret-manager-untrusted` secret containing credentials for a GCP service account with permission to access secrets.

Reviewer: ```suggestion
2. Add the `sa-secret-manager-untrusted` Secret containing credentials for a GCP service account with permission to access Secrets.
```

3. Add the `external-secrets` Helm repository. Use the following command: `helm repo add external-secrets https://external-secrets.github.io/kubernetes-external-secrets/`.
4. Install the `external-secrets/kubernetes-external-secrets` Helm chart. Run `helm install -f prow/cluster/resources/external-secrets/values_untrusted.yaml -n external-secrets kubernetes-external-secrets external-secrets/kubernetes-external-secrets`.

# Configuration

> Secrets can be stored as text in GCP Secret Manager and be mapped to a Kubernetes secret with one key. 

Reviewer: ```suggestion
Secrets can be stored as text in GCP Secret Manager and be mapped to a Kubernetes Secret with one key. 
```

See an example:

```yaml
apiVersion: kubernetes-client.io/v1
kind: ExternalSecret
metadata:
  name: plainSecret # name of the k8s external secret and the k8s secret
spec:
  backendType: gcpSecretsManager
  projectId: my-gcp-project
  data:
    - key: gcp-plain-secret # name of the GCP secret
      name: token # key name in the k8s secret
      version: latest # version of the GCP secret
```

> Secrets can also be stored as JSON in GCP Secret Manager and be mapped to a Kubernetes secret with multiple keys. 

Reviewer: ```suggestion
Secrets can also be stored as JSON in GCP Secret Manager and be mapped to a Kubernetes Secret with multiple keys. 
```

See an example:

```yaml
apiVersion: kubernetes-client.io/v1
kind: ExternalSecret
metadata:

>   name: secretName # name of the k8s external secret and the k8s secret

Reviewer: ```suggestion
  name: secretName # name of the k8s external Secret and the k8s Secret
```

spec:
  backendType: gcpSecretsManager
  projectId: my-gcp-project
  data:

>     - key: gcp-json-secret # name of the GCP secret

Reviewer: ```suggestion
    - key: gcp-json-secret # name of the GCP Secret
```

>       name: keyName # key name in the k8s secret

Reviewer: ```suggestion
      name: keyName # key name in the k8s Secret
```

>       version: latest # version of the GCP secret

Reviewer: ```suggestion
      version: latest # version of the GCP Secret
```

>       property: keyName # name of the field in the GCP secret JSON, unused for plain values

Reviewer: ```suggestion
      property: keyName # name of the field in the GCP Secret JSON, unused for plain values
```

>     - key: gcp-json-secret # name of the GCP secret

Reviewer: ```suggestion
    - key: gcp-json-secret # name of the GCP Secret
```

>       name: anotherKey # key name in the k8s secret

Reviewer: ```suggestion
      name: anotherKey # key name in the k8s Secret
```

>       version: latest # version of the GCP secret

Reviewer: ```suggestion
      version: latest # version of the GCP Secret
```

>       property: anotherKey # name of the field in the GCP secret JSON, unused for plain values

Reviewer: ```suggestion
      property: anotherKey # name of the field in the GCP Secret JSON, unused for plain values
```

```

# External Secrets Checker

> External Secrets Checker checks if all External Secrets have synchronized succesfully, and if all Secrets have corresponding External Secrets.

Reviewer: ```suggestion
External Secrets Checker checks if all External Secrets synchronized successfully, and if all Secrets have corresponding External Secrets.
```

> It can be installed with the `kubectl apply -f external_secrets_checker_prow.yaml` command.

Reviewer: ```suggestion
To install External Secrets Checker run the following command: 

`kubectl apply -f external_secrets_checker_prow.yaml`
```
